packages/big-design/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,153 @@
+import { MoreHorizIcon } from '@bigcommerce/big-design-icons';
+import React, { createRef, HTMLAttributes, memo, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useWindowResizeListener } from '../../hooks';
+import { MarginProps } from '../../mixins';
+import { ButtonProps } from '../Button';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton, StyledFlexItem } from './styled';
+
+export interface ButtonGroupAction extends Omit<ButtonProps, 'children' | 'iconOnly' | 'iconRight' | 'iconLeft'> {
+  text: string;
+  icon?: React.ReactElement;
+}
+
+export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement>, MarginProps {
+  actions: ButtonGroupAction[];
+}
+
+const excludeIconProps = <T extends any>({
+  iconOnly,
+  iconRight,
+  iconLeft,
+  ...actionProps
+}: T): Pick<T, Exclude<keyof T, 'iconOnly' | 'iconRight' | 'iconLeft'>> => actionProps;
+
+export const ButtonGroup: React.FC<ButtonGroupProps> = memo(({ actions, ...wrapperProps }) => {
+  const parentRef = createRef<HTMLDivElement>();
+  const dropdownRef = createRef<HTMLDivElement>();
+  const [isMenuVisible, setIsMenuVisible] = useState(false);
+  const [actionsState, setActionsState] = useState(
+    actions.map((action) => ({ isVisible: true, action: excludeIconProps(action), ref: createRef<HTMLDivElement>() })),
+  );
+
+  const changeActionsVisibility = useCallback(() => {
```

Comment:
🍹 nitpick, but could we align this to a similar name to the `PillsTabs` one?
```suggestion
  const hideOverflowedActions = useCallback(() => {
```

Reply:
Sure, thank you, done ✅